wezterm-gui/src/selection.rs

```diff
@@ -44,6 +44,7 @@ impl Selection {
 pub struct SelectionCoordinate {
     pub x: usize,
     pub y: StableRowIndex,
+    pub treat_zero_in_x_as_before_first_cell: bool,
```

Comment:
Hmm, I'm "worried" that having this be separated from `x` might be prone to bugs when interpreting `x`.

What do you think about changing the type of `x` to reflect this; something like this?

```rust
enum SelectionX {
    /// Zero-based cell index
    Cell(usize),
    /// Before the 0th cell
    BeforeZero,
}
```

```rust
  pub x: SelectionX,
```

The big advantage of this is that the compiler will now be able to tell us where in the code we need to consider these cases.

Reply:
Good idea, done